lib.Makefile

```diff
@@ -0,0 +1,1133 @@
+join_platforms = $(subst $(space),$(comma),$(call prefix_linux,$(strip $1)))
+
+## Targets used when cross building.
+.PHONY: native register
+native:
+ifneq ($(BUILDARCH),$(ARCH))
+	@echo "Target $(MAKECMDGOALS)" is not supported when cross building! && false
+endif
+
+# Enable binfmt adding support for miscellaneous binary formats.
+# This is only needed when running non-native binaries.
+register:
+ifneq ($(BUILDARCH),$(ARCH))
+	docker run --rm --privileged multiarch/qemu-user-static:register || true
+endif
+
+# If this is a release, also tag and push additional images.
+ifeq ($(RELEASE),true)
+PUSH_IMAGES+=$(RELEASE_IMAGES)
+endif
+
+DOCKERHUB_REGISTRY ?=registry.hub.docker.com
+# filter-registry filters out registries we don't want to include when tagging / pushing docker images. For instance,
+# we don't include the registry name when pushing to docker hub because that registry is the default.
+filter-registry ?= $(if $(filter-out $(1),$(DOCKERHUB_REGISTRY)),$(1)/)
+
+# Convenience function to get the first dev image repo in the list.
+DEV_REGISTRY ?= $(firstword $(DEV_REGISTRIES))
+
+# remove from the list to push to manifest any registries that do not support multi-arch
+MANIFEST_REGISTRIES         ?= $(DEV_REGISTRIES)
+
+PUSH_MANIFEST_IMAGES := $(foreach registry,$(MANIFEST_REGISTRIES),$(foreach image,$(BUILD_IMAGES),$(call filter-registry,$(registry))$(image)))
+
+# location of docker credentials to push manifests
+DOCKER_CONFIG ?= $(HOME)/.docker/config.json
+
+# If a repository still relies on vendoring, it must set GOMOD_VENDOR to "true".
+# If that's not the case and we're running in CI, set -mod=readonly to prevent builds
+# from being flagged as dirty due to updates in go.mod or go.sum _except_ for:
+# - for local builds, which _require_ a change to go.mod.
+# - the targets 'commit-pin-updates' and  'golangci-lint' which require
+#   updating go.mod and/or go.sum
+SKIP_GOMOD_READONLY_FLAG =
+ifeq ($(MAKECMDGOALS),commit-pin-updates)
+	SKIP_GOMOD_READONLY_FLAG = yes
+endif
+ifeq ($(MAKECMDGOALS),golangci-lint)
+	SKIP_GOMOD_READONLY_FLAG = yes
+endif
+
+ifeq ($(GOMOD_VENDOR),true)
+	GOFLAGS?="-mod=vendor"
+else
+ifeq ($(CI),true)
+ifndef SKIP_GOMOD_READONLY_FLAG
+	GOFLAGS?="-mod=readonly"
+endif
+endif
+endif
+
+# For building, we use the go-build image for the *host* architecture, even if the target is different
+# the one for the host should contain all the necessary cross-compilation tools
+# we do not need to use the arch since go-build:v0.15 now is multi-arch manifest
+GO_BUILD_IMAGE ?= calico/go-build
+CALICO_BUILD    = $(GO_BUILD_IMAGE):$(GO_BUILD_VER)
+
+# Images used in build / test across multiple directories.
+PROTOC_CONTAINER=calico/protoc:$(PROTOC_VER)-$(BUILDARCH)
+ETCD_IMAGE ?= quay.io/coreos/etcd:$(ETCD_VERSION)-$(ARCH)
+ifeq ($(BUILDARCH),amd64)
+	# *-amd64 tagged images for etcd are not available until v3.5.0
+	ETCD_IMAGE = quay.io/coreos/etcd:$(ETCD_VERSION)
+endif
+
```

Comment:
question: the UBI version "centralization" didn't go in v3.22? (just wondering)

Reply:
seems like not :laughing: 

Did you cherry-pick it?

Reply:
I did not, sorry... I thought it had gone in pre-cutoff, should I do it right now or has the ship sailed?